src/radical/pilot/raptor/master.py

```diff
@@ -289,77 +293,77 @@ def submit_workers(self, descr: Dict[str, Union[str, int]],
         master - the worker ranks are expected to syncronize their ranks as
         needed.
         '''
-        with self._lock:
-
-            tasks = list()
-            base  = self._uid
-
-            cfg            = copy.deepcopy(self._cfg)
-            cfg['info']    = self._info
-            cfg['ts_addr'] = self._task_service.addr
+        tasks = list()
+        base  = self._uid
 
-            for i in range(count):
+        worker_file  = descr.get('worker_file', '')
+        worker_class = descr.get('worker_class', 'DefaultWorker')
 
-                uid        = '%s.worker.%04d' % (base, i)
-                cfg['uid'] = uid
+        del descr['worker_file']
+        del descr['worker_class']
```

Comment:
better to use method `pop`, otherwise `del` with raise an exception if such keys are not in `descr`

```python
        worker_file  = descr.pop('worker_file', '')
        worker_class = descr.pop('worker_class', 'DefaultWorker')
```